_bleio/connection.py

```diff
@@ -0,0 +1,188 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`_bleio.connection`
+=======================================================================
+
+_bleio implementation for Adafruit_Blinka_bleio
+
+* Author(s): Dan Halbert for Adafruit Industries
+"""
+from __future__ import annotations
+from typing import Iterable, Tuple, Union
+
+from bleak import BleakClient
+
+import _bleio.adapter_ as adap
+import _bleio.address
+import _bleio.service
+
+
+Buf = Union[bytes, bytearray, memoryview]
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_Blinka_bleio.git"
+
+
+class Connection:
+    """A BLE connection to another device. Used to discover and interact with services on the other
+    device.
+
+    Usage::
+
+       import _bleio
+
+       my_entry = None
+       for entry in adapter.scan(2.5):
+           if entry.name is not None and entry.name == 'InterestingPeripheral':
+               my_entry = entry
+               break
+
+       if not my_entry:
+           raise Exception("'InterestingPeripheral' not found")
+
+       connection = adapter.connect(my_entry.address, timeout=10)"""
+
+    def __init__(self, address: _bleio.address.Address):
+        """Connections should not be created directly.
+        Instead, to initiate a connection use `Adapter.connect`.
+        Connections may also be made when another device initiates a connection. To use a Connection
+        created by a peer, read the `Adapter.connections` property.
+
+        :param _bleio.address.Address address: _bleio.address.Address of device to connect to
+        """
+        self._address = address
+        self._bleak_client = None
+
+    @classmethod
+    def from_bleak(
+        cls, address: _bleio.address.Address, bleak_client: BleakClient
+    ) -> "Connection":
+        """Create a Connection from bleak information.
+
+        :param ~_bleio.address.Address address: Address of device to connect to
+        :param BleakClient bleak_client: BleakClient used to make connection. (Blinka _bleio only)
+        """
+        conn = Connection(address)
+        conn._bleak_client = bleak_client  # pylint: disable=protected-access
+        return conn
+
+    @property
+    def bleak_client(self):
```

Comment:
Why make this public? Doing so means that code written for RPi won't work in CircuitPython.

Reply:
(People can just access `_bleak_client` if they really want to.)